fl-server/src/handlers.rs

```diff
@@ -356,7 +371,57 @@ pub async fn list_flists_handler(
     Ok(ResponseResult::Flists(flists))
 }
 
-pub async fn flist_exists(dir_path: &std::path::Path, flist_name: &String) -> io::Result<bool> {
+#[utoipa::path(
+	get,
+	path = "/v1/api/fl/preview/{flist_path}",
+	responses(
+        (status = 200, description = "Flist preview result", body = PreviewResponse),
+        (status = 400, description = "Bad request"),
+        (status = 401, description = "Unauthorized user"),
+        (status = 403, description = "Forbidden"),
+        (status = 500, description = "Internal server error"),
+	),
+    params(
+        ("flist_path" = String, Path, description = "flist file path")
+    )
+)]
+#[debug_handler]
+pub async fn preview_flist_handler(
+    Extension(cfg): Extension<config::Config>,
+    Path(flist_path): Path<String>,
+) -> impl IntoResponse {
+    let fl_path = flist_path;
+
+    match validate_flist_path(cfg.users, &cfg.flist_dir, &fl_path).await {
+        Ok(_) => (),
+        Err(err) => return Err(ResponseError::BadRequest(err.to_string())),
+    };
+
+    let content = match unpack_flist(&fl_path).await {
```

Comment:
I think unpacking an flist to do a preview gonna have both bad performance and also waste of disk space. 

Maybe the flist metadata can be walked to generate the preview instead

Reply:
I remember flist object has walk function which receive metadata of the entire file tree in the flist, which then can be used to build a complete representation of the flist without extracting any file. The unpack already uses this to do the unpacking and the rebuilding of the extracted files.